labtool2.0/src/components/StudentTable.js

```diff
@@ -0,0 +1,442 @@
+import React from 'react'
+import { Button, Icon, Table, Form, Popup, Dropdown, Label } from 'semantic-ui-react'
+import { Link } from 'react-router-dom'
+import { connect } from 'react-redux'
+import HorizontalScrollable from './HorizontalScrollable'
+import { getAllTags, tagStudent, unTagStudent } from '../services/tags'
+import { associateTeacherToStudent } from '../services/assistant'
+import { showAssistantDropdown, showTagDropdown, filterByTag, filterByAssistant, selectTeacher, selectTag } from '../reducers/coursePageLogicReducer'
+
+const { Fragment } = React
+
+export class StudentTable extends React.Component {
```

Comment:
This could be functional component

Reply:
see comment in LabtoolComment.js: tl;dr, we should keep these as classes for now and convert all components into functional ones later on